File: VB/GrowthWare.Framework/Model/Profiles/MGroupRoles.vb
Namespace Model.Profiles
    ''' <summary>
    ''' Model object representing GroupRoles
    ''' </summary>
    ''' <remarks></remarks>
    <Serializable(), CLSCompliant(True)>
    Public Class MGroupRoles

        Private m_AddedUpdatedBy As Integer
        Private m_SecurityEntityId As Integer = -1
        Private m_GroupSeqId As Integer = -1
        Private m_Roles As String

        ''' <summary>
        ''' Gets or sets the ADD_UP_BY.
        ''' </summary>
        ''' <value>Account added or updated by.</value>
        Public Property AddedUpdatedBy() As Integer
            Get
                Return m_AddedUpdatedBy
            End Get
            Set(ByVal value As Integer)
                m_AddedUpdatedBy = value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the SE_SEQ_ID.
        ''' </summary>
        ''' <value>The Security Entity ID.</value>
        Public Property SecurityEntityId() As Integer
            Get
                Return m_SecurityEntityId
            End Get
            Set(ByVal value As Integer)
                m_SecurityEntityId = value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the GROUP_SEQ_ID.
        ''' </summary>
        ''' <value>The Group Sequence ID.</value>
        Public Property GroupSeqId() As Integer
            Get
                Return m_GroupSeqId
            End Get
            Set(ByVal value As Integer)
                m_GroupSeqId = value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the roles.
        ''' </summary>
        ''' <value>The roles.</value>
        Public Property Roles() As String
            Get
                Return m_Roles
            End Get
            Set(ByVal value As String)
                If Not String.IsNullOrEmpty(value) Then m_Roles = value.Trim
            End Set
        End Property
    End Class
End Namespace

File: Net_Core/Framework/Models/Table/MGroupRoles.cs
using System;
using GrowthWare.Framework.Models.Base;

namespace GrowthWare.Framework.Models;

/// <summary>
/// Model object representing GroupRoles
/// </summary>
[Serializable(), CLSCompliant(true)]
public class MGroupRoles : AAddedUpdated
{
    /*
     * Though this class inherits from the AbstractBaseModel class, it does not make use of any of the 
     * functions provided by that class only the AddedBy, AddedDate, UpdatedBy, and 
     * UpdatedDate properties.
     */

#region Member Fields
    private int m_AddedUpdatedBy;
    private int m_SecurityEntityID = -1;
    private int m_GroupSeqId = -1;

    private string m_Roles;
#endregion

#region Public Properties

    [DBIgnoreProperty]
    public override string ForeignKeyName => "NOT USED";

    /// <summary>
    /// Gets or Added Updated By id.
    /// </summary>
    /// <value>The Added Updated By id.</value>
    public int AddedUpdatedBy
    {
        get { return m_AddedUpdatedBy; }
        set { m_AddedUpdatedBy = value; }
    }

    [DBIgnoreProperty]
    public override bool IsForeignKeyNumeric => true;

    /// <summary>
    /// Gets or sets the Security Entity Id.
    /// </summary>
    /// <value>The Security Entity Id.</value>
    public int SecurityEntityID
    {
        get { return m_SecurityEntityID; }
        set { m_SecurityEntityID = value; }
    }

    [DBIgnoreProperty]
    public override string TableName => "[ZGWSecurity].[Groups_Security_Entities_Roles_Security_Entities]";

    /// <summary>
    /// Gets or sets the Group Sequence Id.
    /// </summary>
    /// <value>The Group Sequence Id.</value>
    public int GroupSeqId
    {
        get { return m_GroupSeqId; }
        set { m_GroupSeqId = value; }
    }

    /// <summary>
    /// Gets or sets the roles.
    /// </summary>
    /// <value>The roles.</value>
    public string Roles
    {
        get { return m_Roles; }
        set { if (!String.IsNullOrEmpty(value)) m_Roles = value.Trim(); }
    }
#endregion

#region Constructors
    public MGroupRoles()
    {
        this.setDefaults();
    }

    /// <summary>
    /// Initializes a new instance of the <see cref="MGroupRoles"/> class.
    /// </summary>
    /// <param name="roles">comma separated roles</param>
    /// <param name="securityEntityID"></param>
    public MGroupRoles(string roles, int securityEntityID)
    {
        this.setDefaults();
        Roles = roles;
        SecurityEntityID = securityEntityID;
    }

    public MGroupRoles(int groupSeqId, int securityEntityID)
    {
        this.setDefaults();
        GroupSeqId = groupSeqId;
        SecurityEntityID = securityEntityID;
    }
#endregion

    protected override void setDefaults()
    {
        GroupSeqId = -1;
        // base.m_ForeignKeyName = "NOT_USED";
        // m_TableName = "[ZGWSecurity].[Groups_Security_Entities_Roles_Security_Entities]";
    }
}
